omod/src/test/webapp/resources/scripts/navigatorHandlers.js

```diff
@@ -724,12 +725,15 @@ describe("Tests for simple form navigation handlers", function() {
                 firstQuestion.isSelected=true;
                 field.isSelected = true;
                 firstQuestion.fields = [field];
+                spyOn(secondField, 'select');
+                spyOn(secondField, 'toggleSelection');
+                spyOn(secondField, 'isValid').and.returnValue(false);
+                secondQuestion.fields = [ secondField ];
 
                 clickedQuestionHandler(questions, thirdQuestion, event);
 
                 expect(firstQuestion.onExit).not.toHaveBeenCalled();
```

Comment:
I guess I'm going down the rabbit hole here, but I *would* expect firstQuestion.onExit to have been called.
As Dave had mentioned, this whole path is a bit of an edge case, so maybe not worth spending all the time figuring out why this has't been called... perhaps change this to "expect(firstQuestion.onExit).toHaveBeenCalled(); but comment it out and add a TODO to fix in the future?